landoapi/uplift.py

```diff
@@ -143,9 +160,25 @@ def get_uplift_conduit_state(
             f"Every uplifted patch must have an associated bug ID: {missing} do not."
         )
 
+    rev_ids = [revision["id"] for revision in stack_data.revisions.values()]
+    rev_ids_to_all_diffs = get_rev_ids_to_diffs(phab, rev_ids)
+
     stack = RevisionStack(set(stack_data.revisions.keys()), edges)
 
-    return stack_data, stack, target_repo
+    return stack_data, stack, target_repo, rev_ids_to_all_diffs
+
+
+def get_latest_good_binary_diff(diffs: list[dict]) -> dict:
```

Comment:
The method name and docstring can be improved here. "good" doesn't convey any meaning in this context, and referencing a bug in the docstring first line is not ideal. There is no way to tell what this method is trying to do by reading the docstring or method name.